HermesProxy/World/Server/AccountDataManager.cs

```diff
@@ -5,9 +5,68 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Framework;
+using Framework.Logging;
 
 namespace HermesProxy.World.Server
 {
+    public class AccountMetaDataManager
+    {
+        private const string LAST_CHARACTER_FILE = "last_character.txt";
+
+        private readonly string _accountName;
+
+        private string GetAccountMetaDataDirectory()
+        {
+            string path = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+            path = Path.Combine(path, "AccountData");
+            path = Path.Combine(path, _accountName);
+
+            if (!Directory.Exists(path))
+                Directory.CreateDirectory(path);
+
+            return path;
+        }
+        
+        public AccountMetaDataManager(string accountName)
+        {
+            _accountName = accountName;
+        }
+
+        public (string realmName, string charName, ulong charLowerGuid, long lastLoginUnixSec)? GetLastSelectedCharacter()
+        {
+            if (!Settings.RememberLastCharacter)
+                return null;
+
+            var dir = GetAccountMetaDataDirectory();
+            var path = Path.Combine(dir, LAST_CHARACTER_FILE);
```

Comment:
```suggestion
            var path = Path.Combine(GetAccountMetaDataDirectory(), LAST_CHARACTER_FILE);
```

`dir` was only used once, no need for a separate variable